skglm/datafits/single_task.py

```diff
@@ -44,14 +50,18 @@ def params_to_dict(self):
     def initialize(self, X, y):
         self.Xty = X.T @ y
         n_features = X.shape[1]
+        self.global_lipschitz = norm(X, ord=2) ** 2 / len(y)
         self.lipschitz = np.zeros(n_features, dtype=X.dtype)
         for j in range(n_features):
             self.lipschitz[j] = (X[:, j] ** 2).sum() / len(y)
 
-    def initialize_sparse(
-            self, X_data, X_indptr, X_indices, y):
+    def initialize_sparse(self, X_data, X_indptr, X_indices, y):
         n_features = len(X_indptr) - 1
         self.Xty = np.zeros(n_features, dtype=X_data.dtype)
+
+        self.global_lipschitz = spectral_norm(X_data, X_indptr, X_indices, len(y)) ** 2
+        self.global_lipschitz /= len(y)
```

Comment:
As suggested previously, I would add a few comments on why we need the `spectral_norm` function. Like
```
# L2-norm of X not available in the sparse case, `spectral_norm` computes an upper bound on the spectral radius.
```

or something along those lines

Reply:
well no, this time it's the exact spectral norm (beware, not equal to spectral radius. spectral radius is the largest eigenvalue of X, it can be 0 without X being 0)

for me it's explicit enough as is